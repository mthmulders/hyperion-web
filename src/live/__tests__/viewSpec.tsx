import * as React from 'react';

import { shallow } from 'enzyme';

import View from '../view';

describe('<View />', () => {
    it('should present some data', () => {
        // Arrange
        const data = {
            elecCon: 0.265,
            elecProd: 0.000,
            gas: 3781.676,
            tariff: '0002',
            ts: '2018-01-05T15:40:47+01:00',
        };

        // Act
        const wrapper = shallow(<View data={ data } />);

        // Assert
        expect(wrapper).toMatchSnapshot();
    });
});