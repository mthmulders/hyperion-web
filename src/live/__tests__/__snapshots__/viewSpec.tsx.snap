// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<View /> should present some data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <View
    data={
        Object {
            "elecCon": 0.265,
            "elecProd": 0,
            "gas": 3781.676,
            "tariff": "0002",
            "ts": "2018-01-05T15:40:47+01:00",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "allRowsSelected": false,
      "children": <TableBody
        allRowsSelected={false}
        deselectOnClickaway={true}
        displayRowCheckbox={false}
        multiSelectable={false}
        preScanRows={true}
        selectable={true}
        style={Object {}}
>
        <TableRow
                displayBorder={true}
                hoverable={false}
                hovered={false}
                selectable={true}
                striped={false}
        >
                <TableRowColumn
                        hoverable={false}
                >
                        Timestamp
                </TableRowColumn>
                <TableRowColumn
                        hoverable={false}
                >
                        Friday, January 5th 2018, 15:40:47
                </TableRowColumn>
        </TableRow>
        <TableRow
                displayBorder={true}
                hoverable={false}
                hovered={false}
                selectable={true}
                striped={false}
        >
                <TableRowColumn
                        hoverable={false}
                >
                        Current electricity tariff
                </TableRowColumn>
                <TableRowColumn
                        hoverable={false}
                >
                        Normal
                </TableRowColumn>
        </TableRow>
        <TableRow
                displayBorder={true}
                hoverable={false}
                hovered={false}
                selectable={true}
                striped={false}
        >
                <TableRowColumn
                        hoverable={false}
                >
                        Current electricity consumption
                </TableRowColumn>
                <TableRowColumn
                        hoverable={false}
                >
                        0.265 kWh
                </TableRowColumn>
        </TableRow>
        <TableRow
                displayBorder={true}
                hoverable={false}
                hovered={false}
                selectable={true}
                striped={false}
        >
                <TableRowColumn
                        hoverable={false}
                >
                        Current electricity production
                </TableRowColumn>
                <TableRowColumn
                        hoverable={false}
                >
                        0 kWh
                </TableRowColumn>
        </TableRow>
        <TableRow
                displayBorder={true}
                hoverable={false}
                hovered={false}
                selectable={true}
                striped={false}
        >
                <TableRowColumn
                        hoverable={false}
                >
                        Total gas consumption
                </TableRowColumn>
                <TableRowColumn
                        hoverable={false}
                >
                        3781.676 m3
                </TableRowColumn>
        </TableRow>
</TableBody>,
      "fixedFooter": true,
      "fixedHeader": true,
      "height": "inherit",
      "multiSelectable": false,
      "selectable": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "allRowsSelected": false,
        "children": Array [
          <TableRow
            displayBorder={true}
            hoverable={false}
            hovered={false}
            selectable={true}
            striped={false}
>
            <TableRowColumn
                        hoverable={false}
            >
                        Timestamp
            </TableRowColumn>
            <TableRowColumn
                        hoverable={false}
            >
                        Friday, January 5th 2018, 15:40:47
            </TableRowColumn>
</TableRow>,
          <TableRow
            displayBorder={true}
            hoverable={false}
            hovered={false}
            selectable={true}
            striped={false}
>
            <TableRowColumn
                        hoverable={false}
            >
                        Current electricity tariff
            </TableRowColumn>
            <TableRowColumn
                        hoverable={false}
            >
                        Normal
            </TableRowColumn>
</TableRow>,
          <TableRow
            displayBorder={true}
            hoverable={false}
            hovered={false}
            selectable={true}
            striped={false}
>
            <TableRowColumn
                        hoverable={false}
            >
                        Current electricity consumption
            </TableRowColumn>
            <TableRowColumn
                        hoverable={false}
            >
                        0.265 kWh
            </TableRowColumn>
</TableRow>,
          <TableRow
            displayBorder={true}
            hoverable={false}
            hovered={false}
            selectable={true}
            striped={false}
>
            <TableRowColumn
                        hoverable={false}
            >
                        Current electricity production
            </TableRowColumn>
            <TableRowColumn
                        hoverable={false}
            >
                        0 kWh
            </TableRowColumn>
</TableRow>,
          <TableRow
            displayBorder={true}
            hoverable={false}
            hovered={false}
            selectable={true}
            striped={false}
>
            <TableRowColumn
                        hoverable={false}
            >
                        Total gas consumption
            </TableRowColumn>
            <TableRowColumn
                        hoverable={false}
            >
                        3781.676 m3
            </TableRowColumn>
</TableRow>,
        ],
        "deselectOnClickaway": true,
        "displayRowCheckbox": false,
        "multiSelectable": false,
        "preScanRows": true,
        "selectable": true,
        "style": Object {},
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <TableRowColumn
                hoverable={false}
>
                Timestamp
</TableRowColumn>,
              <TableRowColumn
                hoverable={false}
>
                Friday, January 5th 2018, 15:40:47
</TableRowColumn>,
            ],
            "displayBorder": true,
            "hoverable": false,
            "hovered": false,
            "selectable": true,
            "striped": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Timestamp",
                "hoverable": false,
              },
              "ref": null,
              "rendered": "Timestamp",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Friday, January 5th 2018, 15:40:47",
                "hoverable": false,
              },
              "ref": null,
              "rendered": "Friday, January 5th 2018, 15:40:47",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <TableRowColumn
                hoverable={false}
>
                Current electricity tariff
</TableRowColumn>,
              <TableRowColumn
                hoverable={false}
>
                Normal
</TableRowColumn>,
            ],
            "displayBorder": true,
            "hoverable": false,
            "hovered": false,
            "selectable": true,
            "striped": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Current electricity tariff",
                "hoverable": false,
              },
              "ref": null,
              "rendered": "Current electricity tariff",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Normal",
                "hoverable": false,
              },
              "ref": null,
              "rendered": "Normal",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <TableRowColumn
                hoverable={false}
>
                Current electricity consumption
</TableRowColumn>,
              <TableRowColumn
                hoverable={false}
>
                0.265 kWh
</TableRowColumn>,
            ],
            "displayBorder": true,
            "hoverable": false,
            "hovered": false,
            "selectable": true,
            "striped": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Current electricity consumption",
                "hoverable": false,
              },
              "ref": null,
              "rendered": "Current electricity consumption",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "0.265 kWh",
                "hoverable": false,
              },
              "ref": null,
              "rendered": "0.265 kWh",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <TableRowColumn
                hoverable={false}
>
                Current electricity production
</TableRowColumn>,
              <TableRowColumn
                hoverable={false}
>
                0 kWh
</TableRowColumn>,
            ],
            "displayBorder": true,
            "hoverable": false,
            "hovered": false,
            "selectable": true,
            "striped": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Current electricity production",
                "hoverable": false,
              },
              "ref": null,
              "rendered": "Current electricity production",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "0 kWh",
                "hoverable": false,
              },
              "ref": null,
              "rendered": "0 kWh",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <TableRowColumn
                hoverable={false}
>
                Total gas consumption
</TableRowColumn>,
              <TableRowColumn
                hoverable={false}
>
                3781.676 m3
</TableRowColumn>,
            ],
            "displayBorder": true,
            "hoverable": false,
            "hovered": false,
            "selectable": true,
            "striped": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Total gas consumption",
                "hoverable": false,
              },
              "ref": null,
              "rendered": "Total gas consumption",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "3781.676 m3",
                "hoverable": false,
              },
              "ref": null,
              "rendered": "3781.676 m3",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "allRowsSelected": false,
        "children": <TableBody
          allRowsSelected={false}
          deselectOnClickaway={true}
          displayRowCheckbox={false}
          multiSelectable={false}
          preScanRows={true}
          selectable={true}
          style={Object {}}
>
          <TableRow
                    displayBorder={true}
                    hoverable={false}
                    hovered={false}
                    selectable={true}
                    striped={false}
          >
                    <TableRowColumn
                              hoverable={false}
                    >
                              Timestamp
                    </TableRowColumn>
                    <TableRowColumn
                              hoverable={false}
                    >
                              Friday, January 5th 2018, 15:40:47
                    </TableRowColumn>
          </TableRow>
          <TableRow
                    displayBorder={true}
                    hoverable={false}
                    hovered={false}
                    selectable={true}
                    striped={false}
          >
                    <TableRowColumn
                              hoverable={false}
                    >
                              Current electricity tariff
                    </TableRowColumn>
                    <TableRowColumn
                              hoverable={false}
                    >
                              Normal
                    </TableRowColumn>
          </TableRow>
          <TableRow
                    displayBorder={true}
                    hoverable={false}
                    hovered={false}
                    selectable={true}
                    striped={false}
          >
                    <TableRowColumn
                              hoverable={false}
                    >
                              Current electricity consumption
                    </TableRowColumn>
                    <TableRowColumn
                              hoverable={false}
                    >
                              0.265 kWh
                    </TableRowColumn>
          </TableRow>
          <TableRow
                    displayBorder={true}
                    hoverable={false}
                    hovered={false}
                    selectable={true}
                    striped={false}
          >
                    <TableRowColumn
                              hoverable={false}
                    >
                              Current electricity production
                    </TableRowColumn>
                    <TableRowColumn
                              hoverable={false}
                    >
                              0 kWh
                    </TableRowColumn>
          </TableRow>
          <TableRow
                    displayBorder={true}
                    hoverable={false}
                    hovered={false}
                    selectable={true}
                    striped={false}
          >
                    <TableRowColumn
                              hoverable={false}
                    >
                              Total gas consumption
                    </TableRowColumn>
                    <TableRowColumn
                              hoverable={false}
                    >
                              3781.676 m3
                    </TableRowColumn>
          </TableRow>
</TableBody>,
        "fixedFooter": true,
        "fixedHeader": true,
        "height": "inherit",
        "multiSelectable": false,
        "selectable": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "allRowsSelected": false,
          "children": Array [
            <TableRow
              displayBorder={true}
              hoverable={false}
              hovered={false}
              selectable={true}
              striped={false}
>
              <TableRowColumn
                            hoverable={false}
              >
                            Timestamp
              </TableRowColumn>
              <TableRowColumn
                            hoverable={false}
              >
                            Friday, January 5th 2018, 15:40:47
              </TableRowColumn>
</TableRow>,
            <TableRow
              displayBorder={true}
              hoverable={false}
              hovered={false}
              selectable={true}
              striped={false}
>
              <TableRowColumn
                            hoverable={false}
              >
                            Current electricity tariff
              </TableRowColumn>
              <TableRowColumn
                            hoverable={false}
              >
                            Normal
              </TableRowColumn>
</TableRow>,
            <TableRow
              displayBorder={true}
              hoverable={false}
              hovered={false}
              selectable={true}
              striped={false}
>
              <TableRowColumn
                            hoverable={false}
              >
                            Current electricity consumption
              </TableRowColumn>
              <TableRowColumn
                            hoverable={false}
              >
                            0.265 kWh
              </TableRowColumn>
</TableRow>,
            <TableRow
              displayBorder={true}
              hoverable={false}
              hovered={false}
              selectable={true}
              striped={false}
>
              <TableRowColumn
                            hoverable={false}
              >
                            Current electricity production
              </TableRowColumn>
              <TableRowColumn
                            hoverable={false}
              >
                            0 kWh
              </TableRowColumn>
</TableRow>,
            <TableRow
              displayBorder={true}
              hoverable={false}
              hovered={false}
              selectable={true}
              striped={false}
>
              <TableRowColumn
                            hoverable={false}
              >
                            Total gas consumption
              </TableRowColumn>
              <TableRowColumn
                            hoverable={false}
              >
                            3781.676 m3
              </TableRowColumn>
</TableRow>,
          ],
          "deselectOnClickaway": true,
          "displayRowCheckbox": false,
          "multiSelectable": false,
          "preScanRows": true,
          "selectable": true,
          "style": Object {},
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <TableRowColumn
                  hoverable={false}
>
                  Timestamp
</TableRowColumn>,
                <TableRowColumn
                  hoverable={false}
>
                  Friday, January 5th 2018, 15:40:47
</TableRowColumn>,
              ],
              "displayBorder": true,
              "hoverable": false,
              "hovered": false,
              "selectable": true,
              "striped": false,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Timestamp",
                  "hoverable": false,
                },
                "ref": null,
                "rendered": "Timestamp",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Friday, January 5th 2018, 15:40:47",
                  "hoverable": false,
                },
                "ref": null,
                "rendered": "Friday, January 5th 2018, 15:40:47",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <TableRowColumn
                  hoverable={false}
>
                  Current electricity tariff
</TableRowColumn>,
                <TableRowColumn
                  hoverable={false}
>
                  Normal
</TableRowColumn>,
              ],
              "displayBorder": true,
              "hoverable": false,
              "hovered": false,
              "selectable": true,
              "striped": false,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Current electricity tariff",
                  "hoverable": false,
                },
                "ref": null,
                "rendered": "Current electricity tariff",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Normal",
                  "hoverable": false,
                },
                "ref": null,
                "rendered": "Normal",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <TableRowColumn
                  hoverable={false}
>
                  Current electricity consumption
</TableRowColumn>,
                <TableRowColumn
                  hoverable={false}
>
                  0.265 kWh
</TableRowColumn>,
              ],
              "displayBorder": true,
              "hoverable": false,
              "hovered": false,
              "selectable": true,
              "striped": false,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Current electricity consumption",
                  "hoverable": false,
                },
                "ref": null,
                "rendered": "Current electricity consumption",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "0.265 kWh",
                  "hoverable": false,
                },
                "ref": null,
                "rendered": "0.265 kWh",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <TableRowColumn
                  hoverable={false}
>
                  Current electricity production
</TableRowColumn>,
                <TableRowColumn
                  hoverable={false}
>
                  0 kWh
</TableRowColumn>,
              ],
              "displayBorder": true,
              "hoverable": false,
              "hovered": false,
              "selectable": true,
              "striped": false,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Current electricity production",
                  "hoverable": false,
                },
                "ref": null,
                "rendered": "Current electricity production",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "0 kWh",
                  "hoverable": false,
                },
                "ref": null,
                "rendered": "0 kWh",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <TableRowColumn
                  hoverable={false}
>
                  Total gas consumption
</TableRowColumn>,
                <TableRowColumn
                  hoverable={false}
>
                  3781.676 m3
</TableRowColumn>,
              ],
              "displayBorder": true,
              "hoverable": false,
              "hovered": false,
              "selectable": true,
              "striped": false,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Total gas consumption",
                  "hoverable": false,
                },
                "ref": null,
                "rendered": "Total gas consumption",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "3781.676 m3",
                  "hoverable": false,
                },
                "ref": null,
                "rendered": "3781.676 m3",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
